### PR TITLE
feat: remove outdated metadata UI manifests

### PR DESCRIPTION
**Pull request by Bobgy**
**Which issue is resolved by this Pull Request:**
Resolves https://github.com/kubeflow/metadata/issues/250

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

File: metadata/base/kustomization.yaml [hunk 1]
```diff
@@ -43,13 +38,6 @@ vars:
     version: v1
   fieldref:
     fieldpath: data.uiClusterDomain
-- name: metadata-service
```
[the remark this answers is not in the document]
Comment (Bobgy) [reply metadata/base/kustomization.yaml:46]: The obj reference targets UI service, and I removed its usage.

Do we still need to keep metadata server?
If yes, how do people access it? And how do people access it before? Was the UI service a proxy to the server?

**Review by Bobgy — COMMENTED** [review]
Reply

**Comment by PatrickXYS** [comment]
The reason that E2E tests failed is 

https://github.com/kubeflow/kfctl/blob/master/py/kubeflow/kfctl/testing/pytests/kf_is_ready_test.py#L86-L98

```
def test_metadata_is_ready(record_xml_attribute, namespace, cluster_name):
    deployment_names = [
        "metadata-deployment",
        "metadata-grpc-deployment",
        "metadata-db",
        "metadata-ui",
        "metadata-envoy-deployment",
        "metadata-writer",
    ]
    check_deployments_ready(record_xml_attribute, namespace,
                            "test_metadata_is_ready", deployment_names,
                            cluster_name)
```

Since this PR deleted `metadata-ui`, which breaks E2E tests, should send a PR to change E2E as well

File: metadata/base/kustomization.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jeffwan) [reply metadata/base/kustomization.yaml:46]: We discuss this in another thread. I think we all agree to remove this. https://github.com/kubeflow/metadata/issues/250

**Review by Jeffwan — APPROVED** [review]

**Comment by Jeffwan** [comment]
Thanks @Bobgy driving the work

**Comment by PatrickXYS** [comment]
@Bobgy Will this change go into v1.2-branch? If so, we may need cherry-pick this PR as well  https://github.com/kubeflow/kfctl/pull/440

**Comment by Bobgy** [comment]
Yes, I'll cherry pick it

**Comment by PatrickXYS** [comment]
For this PR kubeflow/kfctl#440, @pvaneck helped cherry-pick in this PR https://github.com/kubeflow/kfctl/pull/441

So you should be good to only pick current PR https://github.com/kubeflow/manifests/pull/1619

**Pull request merged**